.github/actions/setup_flutter_environment/action.yml

```diff
@@ -1,32 +1,38 @@
 name: Setup Flutter environment
 description: Setup the Flutter environment
 inputs:
-  prod_release:
-    required: false
-    default: 'false'
-    description: "Is prod release"
-
-env:
-  FLUTTER_VERSION: 3.13.7
+  build_type:
+    required: true
+    description: >
+      Whether to build a development or production version of the app.
+      A 'development' build connects to the development backend for testing,
+      while a 'production' build connects to the production backend.
+      It is an error to specify a build type other than
+      'development' or 'production'.
 
 runs:
   using: "composite"
   steps:
+    - name: Validate build_type
+      uses: ./.github/actions/verify_build_type
+      with:
+        build_type: ${{ inputs.build_type }}
+
     - name: Setup Flutter environment
       uses: subosito/flutter-action@v2.11.0
       with:
-        flutter-version: ${{ env.FLUTTER_VERSION }}
+        flutter-version: 3.16.0
```

Comment:
Fun fact: since using `env` is not valid at the root level of an actions file, fixing this (by moving the flutter version number to this position) caused build errors caused by using an outdated version of Flutter. So, until now, the FLUTTER_VERSION env variable was ignored by the pipeline.

I've updated the version number to match the one in the top-level README.md